packages/atoms/src/Switch.tsx

```diff
@@ -1,55 +1,82 @@
-import React from 'react';
-import { TouchableOpacity } from 'react-native';
+import React, { useEffect } from 'react';
+import Animated, {
+    Easing,
+    useAnimatedStyle,
+    useSharedValue,
+    interpolateColor,
+    withTiming,
+} from 'react-native-reanimated';
 import { prepareNativeStyle, useNativeStyles } from '@trezor/styles';
-import { Box } from './Box';
+import { Pressable } from 'react-native';
 
 type SwitchProps = {
     isChecked: boolean;
     onChange: (value: boolean) => void;
     isDisabled?: boolean; // Functionality of disabled works but styles are not implemented yet (waiting for design)
 };
 
-type SwitchStyleProps = {
-    isChecked: boolean;
-};
-const wrapperStyle = prepareNativeStyle<SwitchStyleProps>((utils, { isChecked }) => ({
-    width: 44,
-    height: 24,
-    justifyContent: 'center',
-    alignItems: 'flex-start',
+const SWITCH_CONTAINER_WIDTH = 44;
+const SWITCH_CONTAINER_HEIGHT = 24;
+const SWITCH_CIRCLE_SIZE = 20;
+const SWITCH_CIRCLE_MARGIN = 2;
+const TRACK_CIRCLE_WIDTH = SWITCH_CONTAINER_WIDTH - SWITCH_CIRCLE_SIZE - SWITCH_CIRCLE_MARGIN * 2;
+
+const switchContainerStyle = prepareNativeStyle(utils => ({
+    width: SWITCH_CONTAINER_WIDTH,
+    height: SWITCH_CONTAINER_HEIGHT,
     backgroundColor: utils.colors.gray400,
     borderColor: utils.colors.forest,
     borderRadius: utils.borders.radii.round,
-    extend: {
-        condition: isChecked,
-        style: {
-            alignItems: 'flex-end',
-            backgroundColor: utils.colors.green,
-        },
-    },
+    flexDirection: 'row',
 }));
 
-const SWITCH_CIRCLE_SIZE = 20;
-
 const switchCircleStyle = prepareNativeStyle(utils => ({
     width: SWITCH_CIRCLE_SIZE,
     height: SWITCH_CIRCLE_SIZE,
     backgroundColor: utils.colors.white,
     borderRadius: utils.borders.radii.round,
-    margin: 2,
+    margin: SWITCH_CIRCLE_MARGIN,
+    alignSelf: 'center',
 }));
 
 export const Switch = ({ isChecked, onChange, isDisabled = false }: SwitchProps) => {
-    const { applyStyle } = useNativeStyles();
+    const { applyStyle, utils } = useNativeStyles();
+    const translateX = useSharedValue(0);
+
+    useEffect(() => {
+        translateX.value = withTiming(isChecked ? 0 : TRACK_CIRCLE_WIDTH, {
```

Comment:
Can you please verify you did everything correct? Animation doesn't feel right and much faster than in Figma, check this demo https://www.figma.com/proto/Z6AGVUmKQzLNtDozFamW7f/s2-Mobile?node-id=245%3A5212&scaling=scale-down&page-id=37%3A756&starting-point-node-id=46%3A1178 try to switch Switch few times and it looks different than in your video

Reply:
Looks much better thanks.

One thing that got my attention is that fps drop around 3s, that should not happen for such simple component :D did you tried bechmark it in Flipper if it runs 60 fps all the time?

Reply:
The video above was recorded from our DemoScreen with many rerendered children. So there are some drops on JS thread - **BUT** this was run in _development mode_ and JS thread performance really suffers running in dev mode. It would be more interesting to test performance when building the app for production is ready - soon I hope :).

https://user-images.githubusercontent.com/2227592/172190505-fe15a72c-2e0b-48a0-b3d7-a80aa53df5af.mov

**...BUT no. 2** - that animation runs smoothly - see both UI and JS threads.
https://user-images.githubusercontent.com/2227592/172190577-cf086153-4704-4e86-8a01-c385c664743f.mov

Reply:
Maybe whole screen is rerender because that switch has changing state on that screen, looks good lets fine tune later